packages/pyolite-kernel/py/pyolite/pyolite/__init__.py

```diff
@@ -14,38 +14,16 @@
 # apply patches for available modules
 ensure_matplotlib_patch()
 
-import pyolite
-
 from .kernel import Pyolite
 
 kernel_instance = Pyolite()
 
+termios_mock = types.ModuleType("termios")
+termios_mock.TCSAFLUSH = 2
 
-# TODO: Make an ipython module mock
-class IPMock:
-    def __init__(self, kernel):
-        self.kernel = kernel
-
-
-class InteractiveShellMock:
-    pass
-
-
-ip_mock = IPMock(kernel_instance)
-
-ip = types.ModuleType("IPython")
-ip.get_ipython = lambda: ip_mock
-ip.InteractiveShell = InteractiveShellMock
-
-pt = types.ModuleType("pylabtools")
-pt.backend2gui = {}
-
-core = types.ModuleType("core")
-core.pylabtools = pt
+fcntl_mock = types.ModuleType("fcntl")
+resource_mock = types.ModuleType("resource")
 
-sys.modules["IPython.display"] = pyolite.display
-sys.modules["IPython"] = ip
-sys.modules["IPython.core"] = core
-sys.modules["IPython.core.getipython"] = ip
```

Comment:
Checking on the RTD preview, it looks like `get_ipython()` would now return `None`:

![image](https://user-images.githubusercontent.com/591645/123252996-36ffbf00-d4ed-11eb-87a0-45880068f6e4.png)

Related to: https://github.com/ipython/ipython/blob/abee6378682179bbd3a57bd11eda65a5ad9cbaae/IPython/core/getipython.py#L17-L24

So we might still need that mock for now?

Reply:
@madhur-tandon Either we continue to mock this one for now and merge the PR as is (so that all examples are working), or we create an InteractiveShell and have get_ipython return it? I guess doing it in two PRs is good 👍🏽 the second pass of implementing the interactiveshell might be a big PR

Reply:
This is fixed now.